### DISCUSSION TITLE
Just want to thank you

### DISCUSSION THREAD

**Discussion by kerunaru** [directvt/vtm#109] [General]
I'm an extensive terminal user: vim, mpd, mutt, irssi… You get the idea. This is a great (and cool) piece of software. I'm on a M1 MacBook Air just in case you want to release a version for Apple Silicon.

Thank you and keep the good work! 💪

**Comment by o-sdn-o** [comment]
Nice to hear that, Thank you for your feedback!

> I'm on a M1 MacBook Air just in case you want to release a version for Apple Silicon.

I added to the releases `Mac OS X Universal Binary` file with two architectures inside
- x86_64
- ARM64

[vtm_macos_any](https://github.com/netxs-group/vtm/releases/latest/download/vtm_macos_any.tar.gz)
[vtm_demo_macos_any](https://github.com/netxs-group/vtm/releases/latest/download/vtm_demo_macos_any.tar.gz)

I hope this build runs on the Apple M1 chip. At the moment, I cannot confirm that it works on this architecture. I would be very grateful if you check it out.

**Reply by kerunaru** [reply]
Nice! I just tested it out and it works fine. Now [Homebrew](https://brew.sh) just works, before I got an incorrect architecture error… I think that's due to run VTM under Rossetta 2.

```
Error: Cannot install in Homebrew on ARM processor in Intel default prefix (/usr/local)!
Please create a new installation in /opt/homebrew using one of the
"Alternative Installs" from:
  https://docs.brew.sh/Installation
You can migrate your previously installed formula list with:
  brew bundle dump
```

Tomorrow I will remain all day at work using ARM64 version. I'll tell you if I find something weird. 👍

**Reply by kerunaru** [reply]
So far, so good. I only notice some kind of [color clash](https://en.wikipedia.org/wiki/Attribute_clash) on the window title:

<img width="645" alt="Captura de pantalla 2021-08-03 a las 19 29 54" src="https://user-images.githubusercontent.com/94023/128076290-adb538d6-b959-4fad-940a-fa46516a3889.png">

Also get a blank line when zsh suggest me autocompletion at the last line on Term:

<img width="853" alt="Captura de pantalla 2021-08-03 a las 19 30 41" src="https://user-images.githubusercontent.com/94023/128076443-7b0f9fbd-7cc3-403f-af0a-dbd158c2146d.png">

If I loop through options, they remain visible once focus.

<img width="871" alt="Captura de pantalla 2021-08-03 a las 19 30 53" src="https://user-images.githubusercontent.com/94023/128076712-8446b7ef-9eea-438a-a8e8-ca3f4de0f26b.png">

**Reply by o-sdn-o** [reply]
Yeah, I'll fix it.

I also think to combine everything into one executable file `vtm`, instead of two `vtm` and `vtmd`.

And I want to rename the project on github from `VTM` to `vtm` (smal letters). ~There will break some links to this project~, but for the future I think it will be worth it.

Thanks for testing and reporting bugs.

**Reply by kerunaru** [reply]
No probs! I'll keep using it and reporting whatever I found.

**Reply by o-sdn-o** [reply]
> some kind of color clash on the window title

By default, the color is set to transparent, and the title is colored the same color on top of which it is displayed. I will replace the transparent color with pure white, which I think will be acceptable.

But for now, using escape sequences, you can colorize the window title, but the truth is, most applications overwrite it every time the title is updated.

`ESC` `]` `2` `;` _Window_Title_ `ESC` `\`

Window title = `\e[97mHi \e[93mThere \e[93m!!! \e[91mColored \e[94mTitle`

```
echo -e '\e]2;\e[97mHi \e[93mThere \e[93m!!! \e[91mColored \e[94mTitle\e\\'; cat
```

![image](https://user-images.githubusercontent.com/11535558/128172078-5e45bff0-fffd-424f-b448-17d4421a81a5.png)

**Reply by o-sdn-o** [reply]
You can set the horizontal alignment of the title

Set text alignment `ESC [` `11` `:` `n` `p`

- n = 0 `default` (default is Left)
- n = 1 `Left`
- n = 2 `Right`
- n = 3 `Center`

_Window_title_=`\e[11:2p\e[97mHi \e[93mThere \e[93m!!! \e[91mColored \e[94mTitle`

```
echo -e '\e]2;\e[11:2p\e[97mHi \e[93mThere \e[93m!!! \e[91mColored \e[94mTitle\e\\';cat
```


![image](https://user-images.githubusercontent.com/11535558/128174151-5871111b-21ed-4eb8-bc91-b894deab4570.png)

You can make multi-line window titles, any text is acceptable there.

Linux:  
_Window_title_=`\e[97mHi \e[93mThere \e[93m!!!\r\f\e[91mColored\r\f\e[96mMulti-line Title\e[m`

`\r\f` is used as a new line

macOS:  
_Window_title_=`\e[97mHi \e[93mThere \e[93m!!!\n\e[91mColored\n\e[96mMulti-line Title\e[m`

`\n` is used as a new line

![image](https://user-images.githubusercontent.com/11535558/128179710-b8faa0dd-445d-4aa3-a1cd-f76759b82324.png)

**Reply by o-sdn-o** [reply]
Sorry for several of the same messages, I was a little mistaken and wrote in the wrong place.

**Reply by o-sdn-o** [reply]
Note, there can be problems with window titles containing escaped sequences if the command (`printf`, `echo`) that outputs them breaks this line into several parts during the output process. In this case, the original (whole) sequence breaks down.

**Reply by kerunaru** [reply]
>  I will replace the transparent color with pure white, which I think will be acceptable.

Is it possible the foreground inverts the background color?  If you make it white, the title will be invisible when the background has the same color. 🤔

**Reply by o-sdn-o** [reply]
> If you make it white, the title will be invisible when the background has the same color

That's right, it is!

I want to make the title either bright white or completely black, depending on how bright the background color is. (this is only if it is set transparent)

**Reply by o-sdn-o** [reply]
> Is it possible the foreground inverts the background color?

If I change the background of the title, then it merges with the window. It turns out a window with jagged edges instead of a rectangular window. I need to experiment some more.

![image](https://user-images.githubusercontent.com/11535558/128188347-3f5baa92-972e-4ae4-9955-2df7644894c5.png)

**Reply by o-sdn-o** [reply]
I added a new shader for rendering the title bar

https://github.com/netxs-group/vtm/blob/8b58d8f69f8acd8a4ce21b123872d3ea7200885b/src/netxs/console/console.hpp#L2768-L2778

the result

https://user-images.githubusercontent.com/11535558/128198497-abe04f2d-3d82-4359-8a66-578f08857646.mp4

**Reply by kerunaru** [reply]
Awesome!! That's just perfect. :D

**Reply by o-sdn-o** [reply]
Test with complex background


https://user-images.githubusercontent.com/11535558/128218474-6898cf72-e2f4-4fee-9224-4205049c89d8.mp4

**Reply by o-sdn-o** [reply]
> Also get a blank line when zsh suggest me autocompletion at the last line on Term.
> If I loop through options, they remain visible once focus.

I'm still investigating this behavior and will fix it later.
